contracts/staking/EpochSelector.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "./interfaces/IEpochSelector.sol";
+import "./ClusterSelector_wih_abi_encode.sol";
+
+/// @title Contract to select the top 5 clusters in an epoch
+contract EpochSelector is AccessControl, ClusterSelector, IEpochSelector {
+    /// @notice Event emitted when Cluster is selected
+    /// @param epoch Number of Epoch
+    /// @param cluster Address of cluster
+    event ClusterSelected(uint256 indexed epoch, address indexed cluster);
+
+    /// @notice length of epoch
+    uint256 public constant epochLength = 4 hours;
+
+    /// @notice timestamp when the selector starts
+    uint256 public immutable startTime;
+
+    /// @notice Number of clusters selected in every epoch
+    uint256 public immutable numberOfClustersToSelect;
+
+    /// @notice clusters selected during each epoch
+    mapping(uint256 => address[]) public clustersSelected;
```

Comment:
This public function should be changed to select clusters in case there are no clusters for past epochs and  return whatever is stored in case selection already happened.